File: Code/Ch1-memory-centric-system-model/bare-bones-model/abstract_system_cpu_fetch.py
#### Processor Fetch action
from abstract_system_constants import *

def fetchInstruction(registers,systemState):
    # get the program counter
    pcr = registers[PC]        
    # get the corresponding instruction
    ir = systemState[pcr]            
    # increment the program counter
    registers[PC]+=1
    return ir    

File: Code/Ch1-memory-centric-system-model/cpp-code/abstract_system_cpu_fetch.cc
#ifdef VERBOSE
#include <iostream>
#endif
#include <vector>
#include <tuple>

//### Processor Fetch action
#include "abstract_system_constants.h"

uint64_t fetchInstruction(std::vector<uint64_t>& registers,std::vector<uint64_t>& systemState){
    
// get the program counter
    uint64_t pcr=registers[PC];
#ifdef VERBOSE
    std::cout <<"fetch PC: "<<pcr<<"\n";
#endif
// get the corresponding instruction
    uint64_t ir=systemState[pcr];
#ifdef VERBOSE
    std::cout <<"fetch IR: "<<ir<<"\n";
#endif
//        print('CODE:',ir)
// increment the program counter
    registers[PC]+=1;
    return ir;


}
